platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/io/RolloverFileOutputStream.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.io;
+
+import static com.swirlds.logging.utils.StringUtils.toPaddedDigitsString;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * {@link RolloverFileOutputStream} is an {@link OutputStream} implementation that supports rollover
+ * size  functionality for the underlying file. Note: It's important to manage the lifecycle of the output
+ * stream properly by calling {@link #flush()} and {@link #close()} methods when finished using it to ensure proper
+ * resource cleanup.
+ */
+public class RolloverFileOutputStream extends OutputStream {
+    protected final Path logPath;
+    private final int maxRollover;
+    private final long maxFileSize;
+    private final boolean append;
+    private long remainingSize;
+    private FileOutputStream outputStream;
+    private final String baseName;
+    private final String extension;
+    private final int indexLength;
+    private int index;
+
+    /**
+     * Creates an {@link RolloverFileOutputStream}. Supporting size based rollover.
+     * Usage:
+     * <pre>
+     *     // Example usage with size-based rollover
+     *     Path logPath = Paths.get("logs", "example.log");
+     *     long maxFileSize = 1024 * 1024; // 1 MB
+     *     boolean append = true;
+     *     int maxRollover = 5; // Maximum number of rolling files
+     *     RolloverFileOutputStream outputStream = new RolloverFileOutputStream(logPath, maxFileSize, append, maxRollover);
+     * </pre>
+     *
+     * @param logPath         path where the logging file is located. Should contain the base dir + the name of the
+     *                        logging file.
+     * @param maxFileSize     maximum size for the file. The limit is checked with best effort
+     * @param append          if true and the file exists, appends the content to the file. if not, the file is rolled.
+     * @param maxRollover     Within a rolling period, how many rolling files are allowed.
+     */
+    protected RolloverFileOutputStream(
+            final @NonNull Path logPath, final long maxFileSize, final boolean append, final int maxRollover) {
+        this.logPath = logPath.toAbsolutePath().getParent();
+        this.append = append;
+        this.maxRollover = maxRollover;
+        this.maxFileSize = maxFileSize;
+        String baseFile = logPath.getFileName().toString();
+        final int i = baseFile.lastIndexOf(".");
```

Comment:
I dislike variables with only one character

Reply:
Changed it to lastDotIndex. Also used a regular if statement for assigning the value